### PR TITLE
Remove VPA for the csi-driver-node-disk and csi-driver-node-file DaemonSets

### PR DESCRIPTION
**Pull request by ialidzhikov**
**How to categorize this PR?**

/area auto-scaling
/kind enhancement
/platform azure

**What this PR does / why we need it**:
Similar to https://github.com/gardener/gardener-extension-provider-gcp/pull/929

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:
N/A

**Release note**:

```noteworthy user
The kube-system/csi-driver-node-disk and kube-system/csi-driver-node-file DaemonSet are no longer scaled by a VerticalPodAutoscaler as they do not really benefit from it. Removing VerticalPodAutoscaler for these components will reduce unnecessary evictions by VPA and will be a mitigation for https://issues.k8s.io/126921.
```

### CODE REVIEW

**Comment by ialidzhikov** [comment]
I am sorry for all the back and forth with PR (marking ready for review and then draft).

tl;dr: It should be safe to remove VPA for the csi node plugin DaemonSet. The PR is ready for review!

**Comment by hebelsan** [comment]
Could your please also check the tests in `valuesprovider_test.go` like you have done for aws https://github.com/gardener/gardener-extension-provider-aws/pull/1179/files#diff-5028df05945570f5a5d0eaa31db99102d9cb2325abbf2780fc209124598230aa?

**Comment by ialidzhikov** [comment]
Hi @hebelsan ,

> Could your please also check the tests in `valuesprovider_test.go` like you have done for aws https://github.com/gardener/gardener-extension-provider-aws/pull/1179/files#diff-5028df05945570f5a5d0eaa31db99102d9cb2325abbf2780fc209124598230aa?

For provider-azureazure, the `vpaEnabled` value is structured differently from the other providers. In https://github.com/gardener/gardener-extension-provider-azure/blob/ef48ef1fcb24d3dc5e14962708109d4a490d5dd2/charts/internal/shoot-system-components/values.yaml#L10-L11 `vpaEnabled` is a global value that in the current master branch is being used by the `shoot-system-components/cloud-controller-manager` and the `shoot-system-components/csi-driver-node` charts. With this PR the `vpaEnabled` value is still being used by the `shoot-system-components/cloud-controller-manager` charts: https://github.com/gardener/gardener-extension-provider-azure/blob/9bbe6e34936fc9ab47c49e42fe607ea5105cd18f/charts/internal/shoot-system-components/charts/cloud-controller-manager/templates/cloud-node-manager.yaml#L99-L117

That's why this PR does not remove the logic for passing the` vpaEnabled` value: https://github.com/gardener/gardener-extension-provider-azure/blob/e57186f733b6ae868ebbe088bd45887a2c9f3388/pkg/controller/controlplane/valuesprovider.go#L746-L748

Hence, the corresponding unit tests are not adapted.

If you wish, I could adapt the `vpaEnabled` to be specific to the `shoot-system-components/cloud-controller-manager` chart as it is now the only consumer of this value - there is no longer need to have it as global value.

**Comment by ialidzhikov** [comment]
> If you wish, I could adapt the `vpaEnabled` to be specific to the `shoot-system-components/cloud-controller-manager` chart as it is now the only consumer of this value - there is no longer need to have it as global value.

I pushed https://github.com/gardener/gardener-extension-provider-azure/pull/1046/commits/51379c10ba4de04a5afe1fe13474e055cac860ff.

**Review by hebelsan — APPROVED** [review]
/lgtm
